R/github-action.R

```diff
@@ -22,15 +22,19 @@ use_github_action <- function(name,
                               ignore = TRUE,
                               open = FALSE) {
 
-  # Append a `.yaml` extension if needed
-  stopifnot(is_string(name))
+  # Check if a custom URL is being used.
+  if (is.null(url)) {
+    stopifnot(is_string(name))
 
-  if (!grepl("[.]yaml$", name)) {
-    name <- paste0(name, ".yaml")
-  }
+    # Append a `.yaml` extension if needed
+    name = gsub("[.]yml$", ".yaml", name)
+    if (!grepl("[.]yaml$", name)) {
```

Comment:
If you keep it, please switch to `name <-` for assignment, instead of `name =`.

But it feels like it would be simpler to, instead, use regex of `[.]ya?ml$` inside the `if()` (caveat: untested but I assume you see what I mean).

I don't see how we can just substitute `.yaml` for `.yml` if that is not the extension used by the workflow file. We can just allow both `.yaml` and `.yml` through, yeah?

Reply:
@jennybc with @jimhester's scoping, the file extension is a pretty specific case to what is listed here: 

https://github.com/r-lib/actions/tree/master/examples

So, the following URL would work:

https://raw.githubusercontent.com/r-lib/actions/master/examples/pkgdown.yaml

But, this wouldn't:

https://raw.githubusercontent.com/r-lib/actions/master/examples/pkgdown.yml

As a result, I think a two-pass solution is only viable, where the first pass removes replaces `.yml` with `.yaml` if present and the second pass verifies the extension is present. 

Sorry, about the `=` assignment.

Reply:
I don't think this extra logic is necessary, the examples in r-lib/actions (which is the only place that the 'name' parameter will be used) are always going be `.yaml`.

Reply:
I would revert df603d0adfbcf52c4b44676a8cb3a6f4d9962fef

Reply:
Okay, reverted. Thanks @jimhester !